packages/GA4Client/src/common.js

```diff
@@ -64,6 +64,31 @@ Common.prototype.truncateEventAttributes = function (eventAttributes) {
     );
 };
 
+Common.prototype.standardizeParameters = function (parameters) {
+    var standardizedParameters = {};
+    Object.keys(parameters).forEach(key => {
+        var standardizedKey = (' ' + key).slice(1);
+
+        standardizedKey = standardizedKey.replace("[^a-zA-Z0-9_]".toRegex(), "_")
+        for (forbiddenPrefix in forbiddenPrefixes) {
+            if (standardizedKey.startsWith(forbiddenPrefix)) {
+                standardizedKey = standardizedKey.replace(forbiddenPrefix.toRegex(), "")
+            }
+        }
+        while (standardizedKey.isNotEmpty() && !Character.isLetter(standardizedKey.toCharArray()[0])) {
```

Comment:
`isNotEmpty` and `Character` are not valid JS. I think we have an `isEmpty` function you can use instead. As for `isLetter`, you might be able to use a regex or some other process to check if something is in fact a letter.

Reply:
https://stackoverflow.com/a/32567789

I really like this answer (which is second) because it is more readable than regex.  I'd suggest using this function to determine if it is a letter or not

Reply:
So in our case we only want to accept english characters. So this is less acceptable a solution